### PR TITLE
master

### PR DESCRIPTION
**Pull request by Lexsus**
add push-notification server and client

### CODE REVIEW

File: notify-server/src/main/java/com/lexsus/ariaddna/client/AutoConnectSocket.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * Basic Echo Client Socket
+ */
+@WebSocket(maxTextMessageSize = 64 * 1024)
+public class AutoConnectSocket
+{
+    private final CountDownLatch closeLatch;
+    private ClientSocketListener listener;
+    private Session session;
+
+//    public AutoConnectSocket()
```
Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/client/AutoConnectSocket.java]: Please remove dead code

File: notify-server/src/main/java/com/lexsus/ariaddna/client/AutoConnectSocket.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        this.closeLatch.countDown(); // trigger latch
+        listener.onClose(statusCode,reason);
+    }
+
+    @OnWebSocketConnect
+    public void onConnect(Session session)
+    {
+        System.out.printf("Got connect: %s%n",session);
+        this.session = session;
+//        try
```
Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/client/AutoConnectSocket.java]: Please remove dead code

File: notify-server/src/main/java/com/lexsus/ariaddna/client/AutoConnectSocket.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
+package com.lexsus.ariaddna.client;
+
+/**
+ * Created by Vladislav Gasanov on 28.03.2017.
```
Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/client/AutoConnectSocket.java]: May be remove created by or set created by  Lexus? :)

File: notify-server/src/main/java/com/lexsus/ariaddna/client/ClientMessageSystem.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    private boolean   isStopped;
+    public void runMessagesSystem() {
+        isStopped = false;
+        try {
+            clientService.start();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        new Thread(() -> {
```
Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/client/ClientMessageSystem.java]: How many thread you create here?
How you managed thread? 
And why you don't use standard TheardPoool?

File: notify-server/src/main/java/com/lexsus/ariaddna/client/ConfigClient.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        return new PushClientConsumeImpl<>(queue(),  processor());
+    }
+
+    @Bean
+    public ClientMessageSystem clientSystem() {
+        return new ClientMessageSystem(consumer());
+    }
+
+    @Bean
+    public ClientSocketService clientPushNotification() {return new ClientSocketService("ws://127.0.0.1:8080/echo",queue());}
```
Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/client/ConfigClient.java]: Why use hardcode web socket URL? Are you sure?

File: notify-server/src/main/java/com/lexsus/ariaddna/client/Client.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
+package com.lexsus.ariaddna.client;
```
Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/client/Client.java]: Please change packet name to "com.stnetix.ariaddna.pushnotification.client"

File: notify-server/src/main/java/com/lexsus/ariaddna/client/ClientMessageProcessor.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.lexsus.ariaddna.client;
+
+import com.lexsus.ariaddna.server.MessageProcessor;
```
Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/client/ClientMessageProcessor.java]: Please change packet name to "com.stnetix.ariaddna.pushnotification.server.MessageProcessor"

File: notify-server/src/main/java/com/lexsus/ariaddna/server/MyAdvancedEchoCreator.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
+package com.lexsus.ariaddna.server;
+
+import org.eclipse.jetty.websocket.servlet.ServletUpgradeRequest;
+import org.eclipse.jetty.websocket.servlet.ServletUpgradeResponse;
+import org.eclipse.jetty.websocket.servlet.WebSocketCreator;
+
+/**
+ * Created by Lexsus on 01.04.2017.
+ */
+public class MyAdvancedEchoCreator implements WebSocketCreator
```
Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/server/MyAdvancedEchoCreator.java]: Hmm... Are you sure what this classname is matches?
May be change to AriadddnaWebSocketCreator or PushnotifyWebSocketCreator.

File: notify-server/src/main/java/com/lexsus/ariaddna/server/ToUpperWebSocketServlet.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+import org.eclipse.jetty.websocket.servlet.WebSocketCreator;
+import org.eclipse.jetty.websocket.servlet.WebSocketServlet;
+import org.eclipse.jetty.websocket.servlet.WebSocketServletFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+
+@WebServlet(name= "My test WS", urlPatterns="/echo")
```
Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/server/ToUpperWebSocketServlet.java]: It is correct parameters "(name= "My test WS", urlPatterns="/echo")"?

File: notify-server/src/main/java/com/lexsus/ariaddna/server/ToUpperWebSocketServlet.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+        super.destroy();
+    }
+
+
+
+    @Override
+    public void configure(WebSocketServletFactory factory) {
+
+        //TODO run when server has connection
+        factory.getPolicy().setIdleTimeout(100000);
```
Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/server/ToUpperWebSocketServlet.java]: Please use for hardcode parameters constants.

File: notify-server/src/main/java/com/lexsus/ariaddna/client/ClientMessageSystem.java (hunk 4 above)
Comment (Lexsus) [inline notify-server/src/main/java/com/lexsus/ariaddna/client/ClientMessageSystem.java]: I create 2 thread. First - for producer, second - for consumer.
I managed threads with variable - isStopped.
I thought that this task is simple, and it does not need TheardPoool.  Maybe I'm wrong.

Comment (kosmoflyko) [inline notify-server/src/main/java/com/lexsus/ariaddna/client/ClientMessageSystem.java]: I that use this code can make issue in the future. But I think that we can use it's and fix in next releases.

**Comment by kosmoflyko** [comment]
fix #47

**Comment by kosmoflyko** [comment]
fix #48

**Pull request merged**